### PR TITLE
libmgba: rename from mgba

### PR DESCRIPTION
**Pull request by FnControlOption**
Instruct users to install the cask if they want mGBA.app

This PR should be merged before https://github.com/Homebrew/homebrew-cask/pull/109040

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by alebcay** [comment]
I would advise a formula rename to `libmgba` to make it more clear that this formula contains only the library while the full-featured emulator is now packaged as a Cask.

File: Formula/libmgba.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    <<~EOS
+      This formula only installs libmgba.
+
+      mGBA.app can be downloaded directly from the website:
+        https://mgba.io/
+
+      Alternatively, install with Homebrew Cask:
+        brew install --cask mgba
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/libmgba.rb:37]: i think this is clear enough from the rename

**Comment by carlocab** [comment]
Is there any problem with the original formula?

I know we try not to install app bundles as formulae, but I think now that we have, there's not really much reason to change it unless there's actually a problem with it. If everything works fine, we're just forcing changes on this formula's users for no real reason.

**Comment by nandahkrishna** [comment]
> Is there any problem with the original formula?

The formula actually doesn't work on Catalina and newer, `mgba` crashes when run. I had a local patch to change the build but I didn't submit a PR because it still doesn't fix the formula. See https://github.com/mgba-emu/mgba/issues/1754.

**Comment by carlocab** [comment]
We should have a bunch of other formulae that link with OpenGL -- are those broken too?

**Comment by FnControlOption** [comment]
From https://github.com/mgba-emu/mgba/issues/2019#issuecomment-765725434

> This happens because Apple decided to break OpenGL on threads entirely, which mGBA depends on.

**Review by MikeMcQuaid — COMMENTED** [review]
Fine with me!

**Pull request closed without merging**